serializers/src/glTF/2.0/babylon.glTFMaterial.ts

```diff
@@ -997,7 +941,7 @@ module BABYLON.GLTF2 {
                 return null;
             }
 
-            const metallicRoughnessFactors = this._ConvertSpecularGlossinessTexturesToMetallicRoughness(babylonPBRMaterial.albedoTexture, babylonPBRMaterial.reflectivityTexture, specGloss, mimeType);
+            const metallicRoughnessFactors = this._ConvertSpecularGlossinessTexturesToMetallicRoughness(babylonPBRMaterial.albedoTexture as Texture, babylonPBRMaterial.reflectivityTexture as Texture, specGloss, mimeType);

```

Comment:
Are these textures guaranteed to be `Texture` and not something else that derives from `Texture`? Something to think about.

Reply:
Gotcha, I'll refactor this

Reply:
Just made the refactor to remove casts from BaseTexture to Texture